core/models.py

```diff
@@ -397,6 +397,33 @@ def __str__(self):
         return f"{self.dept} {self.catalog_number}-{self.section_number}"
 
 
+class UserCourses(models.Model):
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
+    course = models.ForeignKey(Course, on_delete=models.CASCADE)
+    yearterm = models.CharField(max_length=5, blank=False)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def display_yearterm(self):
+        if self.yearterm is None:
+            return ""
+        year_str = self.yearterm[:4]
+        term_str = self.yearterm[4:]
+        term_name = ""
+        if term_str == "1":
+            term_name = "Winter"
+        elif term_str == "3":
+            term_name = "Spring"
+        elif term_str == "4":
+            term_name = "Summer"
+        elif term_str == "5":
+            term_name = "Fall"
```

Comment:
Maybe turn this into a dictionary, so it's easier to see the associations (less verbose)?

```
term_map = {"1": "Winter",
 "3": "Spring",
 "4": "Summer",
 "5": "Fall"}
try:
    term_name = term_map[term_str]
except KeyError:
    ...
```